src/telemetry/telemetry.c

```diff
@@ -290,9 +290,11 @@ telemetry_connect(void)
 
 	if (ret < 0)
 	{
+		ereport(ERROR,
+				(errcode(ERRCODE_INTERNAL_ERROR),
+				 errmsg("could not make a connection to %s", TELEMETRY_ENDPOINT),
+				 errdetail("%s", connection_get_and_clear_error(conn))));
 		connection_destroy(conn);
```

Comment:
My understanding is if you throw an error, you don't need to call `connection_destroy`?

Reply:
indeed, will fix.